datafusion/physical-expr/src/expressions/binary/kernels.rs

```diff
@@ -125,10 +140,42 @@ pub(crate) fn bitwise_shift_right(left: ArrayRef, right: ArrayRef) -> Result<Arr
             binary_bitwise_array_op!(
                 left,
                 right,
-                |a: i64, b: i64| a.wrapping_shr(b as u32),
+                |a: i64, b: i64| a.wrapping_shr((b as u64).try_into().unwrap()),
```

Comment:
Since this function returns a result, I think it would be better if this function returned an error on overflow, rather than `panic`ing (aka can we avoid calling the `unwrap`?)